app/src/main/java/org/schabi/newpipe/info_list/dialog/StreamDialogEntry.java

```diff
@@ -0,0 +1,101 @@
+import org.schabi.newpipe.extractor.stream.StreamType;
+import org.schabi.newpipe.local.history.HistoryRecordManager;
+import org.schabi.newpipe.player.playqueue.SinglePlayQueue;
+import org.schabi.newpipe.util.ExtractorHelper;
+import org.schabi.newpipe.util.NavigationHelper;
+
+import java.util.function.Consumer;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+public class StreamDialogEntry {
+
+    @StringRes
+    public final int resource;
+    @NonNull
+    public final StreamDialogEntryAction action;
+
+    public StreamDialogEntry(@StringRes final int resource,
+                             @NonNull final StreamDialogEntryAction action) {
+        this.resource = resource;
+        this.action = action;
+    }
+
+    public String getString(@NonNull final Context context) {
+        return context.getString(resource);
+    }
+
+    public interface StreamDialogEntryAction {
+        void onClick(Fragment fragment, StreamInfoItem infoItem);
+    }
+
+    public static void openChannelFragment(@NonNull final Fragment fragment,
+                                            @NonNull final StreamInfoItem item,
+                                            final String uploaderUrl) {
+        // For some reason `getParentFragmentManager()` doesn't work, but this does.
+        NavigationHelper.openChannelFragment(
+                fragment.requireActivity().getSupportFragmentManager(),
+                item.getServiceId(), uploaderUrl, item.getUploaderName());
+    }
+
+    /**
+     * Fetches a {@link StreamInfoItem} if it is incomplete and executes the callback.
+     * <br />
+     * This method is required if the info has been fetched
+     * via a {@link org.schabi.newpipe.extractor.feed.FeedExtractor}.
+     * FeedExtractors provide a fast and lightweight method to fetch info,
+     * but the info might be incomplete
+     * (see {@link org.schabi.newpipe.local.feed.service.FeedLoadService} for more details).
+     * @param context
+     * @param item the item which is checked and eventually loaded completely
+     * @param callback
+     */
+    public static void fetchItemInfoIfSparse(@NonNull final Context context,
+                                             @NonNull final StreamInfoItem item,
+                                             @NonNull final Consumer<SinglePlayQueue> callback) {
+        if (!(item.getStreamType() == StreamType.LIVE_STREAM
+                || item.getStreamType() == StreamType.AUDIO_LIVE_STREAM)
+                && item.getDuration() < 0) {
+            // Sparse item: fetched by fast fetch
+            ExtractorHelper.getStreamInfo(
+                    item.getServiceId(),
+                    item.getUrl(),
+                    false
+            )
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(result -> {
```

Comment:
Note: the result of this `subscribe` is never added to a `Disposable` which can then be cleared on exit by the fragment. But don't to anything in this PR, as this problem has always been there.

Maybe we should create an app-wide static disposable (handled by the `App` class) for things like this? It should obviously not be used for things that do not make sense after the fragment related to them is terminated (e.g. loading data into a view, but if the view is destroyed before the loading could finish, the loading should stop, i.e. the disposable generated by the loading process should be disposed). It would instead be used for things that are not related to any fragment/activity but just to the app, such as adding streams to the play queue, opening the channel fragment from the stream info item, ...

Reply:
This sounds like a good idea. We need to open an issue for this.

Reply:
#7942